cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -331,20 +332,22 @@ public void initialize(MasterEnvironmentContext context) throws IOException, Ill
 
     String twillRunnables = context.getConfigurations().get(TWILL_RUNNER_SERVICE_MONITOR_DISABLE);
     boolean enableMonitor = true;
-    if (twillRunnables != null && podLabels != null) {
+    if (twillRunnables != null) {
       for (String twillRunnable : twillRunnables.split(",")) {
         if (podLabels.containsKey(DEFAULT_CONTAINER_LABEL) &&
           podLabels.get(DEFAULT_CONTAINER_LABEL).equals(twillRunnable)) {
           enableMonitor = false;
         }
       }
     }
+    boolean enableKubeJobCleaner = podLabels.containsKey(APPFABRIC_POD_LABEL);
```

Comment:
This is kind of ugly. Can we use the `MasterEnvironment.getTask()` and `MasterEnvironmentTask` instead? That is only executed in app-fabric.

Reply:
Updated the change to use MasterEnvironmentTask instead.